docs/source/specs/openapi.yml

```diff
@@ -729,6 +771,135 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /api/v1/reports/costs/:
+    get:
+      summary: Query to obtain cost reports
+      operationId: getCostReports
+      tags:
+        - Report
+      parameters:
+        - in: header
+          name: Authorization
+          description: e.g. Token 45138a913da44ab89532bab0352ef84b
+          schema:
+            type: string
+          required: true
+        - in: query
+          name: group_by
+          schema:
+            $ref: '#/components/schemas/ReportGroupingCost'
+          description: The grouping to apply to the cost report. No grouping by default.
+        - in: query
+          name: resolution
+          schema:
+            $ref: '#/components/schemas/ReportResolution'
+          description: The resolution to apply to the cost report. Default resolution is daily.
+        - in: query
+          name: time_scope
+          schema:
+            $ref: '#/components/schemas/ReportTimeScope'
+          description: The time scope to apply to the cost report. Default resolution is '-10', meaning the last 10 days. Last 30 days would be '-30'. '-10' and '-30' are only valid when used with 'daily' resolution.'-1' is used for current month, '-2' is used for last month when resolution is 'monthly' otherwise invalid.
+      responses:
+        '200':
+          description: A report object
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ReportCost'
+        '401':
+          description: Unauthorized
+        '500':
+          description: Unexpected Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/v1/reports/instances/:
```

Comment:
How will this scale to include the full array of services?  I'm worried this is a bit too EC2-centric. How will this work for reports on e.g. Lambda or OpsWorks or SES?

What do you think about having the endpoint be `/api/v1/reports/inventory` with a `service` or `product` parameter (or kwarg passed as a dict to a `filter` param) to specify which pieces of the inventory to return?

Reply:
I'm ok with change to `inventory`. I suppose we have a default understanding that we will only return instances if no filter is provided for now (since that is the current requirement)?